### PR TITLE
Check in/49406/pre check in data persist

### PR DESCRIPTION
**Pull request by brianseek**
## Description
This PR adds reload functionality to pre-check-in. Also refactors the AppWrapper into the ReloadWrapper, the wrapper is now added conditionally to page where we want reload capability. This deprecates the old `isValidPage` helper that the wrapper utilized. 

This also unifies the `useCheckInData` hook to be used to fetch data by both apps. `useCheckInData` also gets a few updates to communicate back when data fetch is complete. 

Other chores include removing the loading state from confirmable pages, since it was never really utilized in the app. This is to prevent some of the loading-indicator errors in the console.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#49406

## Testing done
Adds reload tests and updates others.

### CODE REVIEW

File: src/applications/check-in/pre-check-in/pages/Introduction/index.jsx [hunk 1]
```diff
@@ -1,109 +1,55 @@
...
-            }
-
-            // hide loading screen
-            setIsLoading(false);
-          })
-          .catch(() => {
-            goToErrorPage('?error=error-fromlocation-precheckin-introduction');
-          });
+      if (checkInDataError) {
+        goToErrorPage();
```
Comment (brianseek) [inline src/applications/check-in/pre-check-in/pages/Introduction/index.jsx:41]: add param `goToErrorPage('?error=error-fromlocation-precheckin-introduction');`

File: src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.cypress.spec.js [hunk 2]
```diff
@@ -20,37 +20,76 @@ describe('Pre-Check In Experience', () => {
...
       ValidateVeteran.validateVeteran();
       ValidateVeteran.attemptToGoToNextPage();
+    });
+    afterEach(() => {
+      cy.window().then(window => {
+        window.sessionStorage.clear();
+      });
+    });
+
+    it.skip('Introduction page', () => {
```
Comment (wullaski) [inline src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.cypress.spec.js:42]: Can we remove this?

File: src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.error.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      ValidateVeteran.validateVeteran();
+      ValidateVeteran.attemptToGoToNextPage();
+    });
+    afterEach(() => {
+      cy.window().then(window => {
+        window.sessionStorage.clear();
+      });
+    });
+
+    it.skip('Introduction page', () => {
```
Comment (wullaski) [inline src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.error.cypress.spec.js:43]: Remove?

**Review by wullaski — APPROVED** [review]
LGTM! only a nit on leaving commented out code.

**Review by acrollet — COMMENTED** [review]
code-wise this looks good, I think I want to take it through a spin for day-of and pre-check-in on a review environment tomorrow morning just cause there are a lot of changes.

File: src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.cypress.spec.js (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.cypress.spec.js:42]: yes!

Comment (brianseek) [reply src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.cypress.spec.js]: removed

File: src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.error.cypress.spec.js (hunk 3 above)
Comment (brianseek) [reply src/applications/check-in/pre-check-in/tests/e2e/app-reload/reload.page.error.cypress.spec.js]: removed

**Pull request merged**